### ISSUE TITLE
Location of user-added words

### ISSUE THREAD

**Issue by gkvallis** [streetsidesoftware/vscode-spell-checker#208]
In the documentation it says that words added by the user will be added to the file cSpell.json that is in the .vscode directory. However, on my Mac it simply adds the new words to my user settings, that is to the file ~/Library/Application Support/Code/User/settings.json.  The spell checker adds entries like 
  "cSpell.userWords": [
        "boussinesq",
       "condensational",
       etc
      ],
This pollutes the settings.json file, since I have >> 100 words. 

Is this expected behaviour or is it a bug? How do I make it add words to the cSpell.json file instead? I tried creating such a file manually but it still didn't add words to it. 
Thanks

**Comment by Jason3S** [comment]
There are three different things going on:

1. The docs are a bit out of date. When they were written, it was not possible to have VS Code update the settings. So, a separate `cSpell.json` had to be used, which VS Code loaded automatically. This behavior changed a while back. VS Code no longer loads the extension files automatically.
2. To avoid polluting the user's repositories, the `cSpell.json` file is no longer automatically created for each workspace / folder.
3. If you create a `cSpell.json` file in your repository, then this extension will add words to it if you add them to `the folder dictionary`.

It is a bit confusing.

**Comment by gkvallis** [comment]
Can you clarify what you mean by 'your repository'? Where exactly do I create this file, on a Mac? I tried putting a cSpell.json in .vscode, and in ~/Library/Application Support/Code/User, but neither had any effect.

**Comment by Jason3S** [comment]
There currently isn't support for a global cSpell.json where the extension will add words. The challenge is where to put it.

This is related to #61 and #166.

**Comment by gkvallis** [comment]
Well, you could just put the global cSpell.json in .vscode, or in ~/Library/Application Support/Code/User/.  Either would work fine. It would certainly be better than the settings.json file. 

But even for an individual file, let's say ~/texfiles/myfile.tex,  where do I put the cSpell.json file that words will be added to instead of settings.json? Or are you saying this currently is not possible?

It is starting to sound like this is an unresolved issue that many people (e.g #61 and #166) would like to see a solution to.

**Comment by Jason3S** [comment]
The tl;dr: put the `cspell.json` file in the same directory as your file.

The spell checker walks through a series of steps looking for the `cspell.json` configuration files.

In addition to the VS Code configuration, `cspell.json` configuration files are applied from global to local.

For global settings, the spell checker uses [configstore](https://github.com/yeoman/configstore). If the  file exists in the configstore location, it will read it, but it will not write to it. 

Much of the local file behavior is dependent upon VS Code.
If you opened an individual file, not a folder or workspace, then the spell checker will search the same location as the file for the following files: `cspell.json`, `.cspell.json`, `.vscode/cSpell.json`.

If you have a folder opened, it will search the folder for the same set of files.

If you have a workspace opened, it will search the folder of the file currently being checked.

**Comment by jlanza** [comment]
I'm facing quite strange behaviour related to that. If I create a cSpell.json file under my .vscode no spell check is run. However if I populate my .vscode/settings.json with the same content as the cSpell.json (adapting json keys) it works.

How does the cSpell.json works? BTW I'm using the VSCode 1.29.

This is my cSpell.json file just in case something is wrong.
    {
      "version": "0.1",
      "language": "es",
      "dictionaryDefinitions": [
        { "name": "projectDictionary", "path": "./dictionary.txt"}
      ],
      "dictionaries": [ "projectDictionary" ],
      "languageSettings": [
        { "languageId": "*", "dictionaries": [ "projectDictionary" ] }
      ],
    }

**Comment by Jason3S** [comment]
Closing stale issues.
